algorithms/48/fib.cpp

```diff
@@ -1,17 +1,16 @@
 #include <iostream>
 using namespace std;
 
-// returns the nth fibonacci number
-long long fib(int n) {
-    if(n == 0) return 0;
-    if(n == 1) return 1;
-    return fib(n-1)+fib(n-2);
-}
 
 int main() {
     int n;
     cout << "Enter a number: ";
     cin >> n;
-    cout << "The " << n << "th fibonacci number is " << fib(n) << endl;
+	long long fib[n+1];
+	fib[1]=0;
+	fib[2]=1;
+	for(int i=3;i<=n;i++)
+		fib[i]=fib[i-1]+fib[i-2];
```

Comment:
You have removed the fib function entirely here. It's better to keep the ```fib``` function and implement your optimization in ```fib```, this leads to cleaner and modular code.
Also you have inconsistant indentation here, please maintain the same indentation as the original file.

Reply:
Will do it soon